File: app/src/main/java/com/example/aaron/recipeassistant/Model/Recipe.java
package com.example.aaron.recipeassistant.Model;


import java.io.Serializable;
import java.util.ArrayList;

public class Recipe implements Serializable {


    private String[] ingredients;
    private String[] directions;
    private String title;
    private String imageUrl;

    public Recipe(String title, String[] ingredients, String[] directions, String imageUrl) {
        this.title = title;
        this.ingredients = ingredients;
        this.directions = directions;
        this.imageUrl = imageUrl;
    }

    public String getTitle() {
        return title;
    }

    public String[] getIngredients() {
        return ingredients;
    }

    public String[] getDirections() {
        return directions;
    }

    public String getImageUrl() {return imageUrl;}

}

File: app/src/main/java/com/example/aaron/recipeassistant/model/Recipe.kt
package com.example.aaron.recipeassistant.model

import android.os.Parcel
import android.os.Parcelable

data class Recipe(val title: String, val ingredients: List<String>, val directions: List<String>, val imageUrl: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.createStringArrayList(),
            parcel.createStringArrayList(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(title)
        parcel.writeStringList(ingredients)
        parcel.writeStringList(directions)
        parcel.writeString(imageUrl)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Recipe> {
        override fun createFromParcel(parcel: Parcel): Recipe {
            return Recipe(parcel)
        }

        override fun newArray(size: Int): Array<Recipe?> {
            return arrayOfNulls(size)
        }
    }
}